app/controllers/restaurants_controller.rb

```diff
@@ -7,7 +7,7 @@ def index
 
   def show
     @restaurant = Restaurant.find(params[:id])
-    @review = Review.new
+    @review ||= Review.new
```

Comment:
What is the purpose of the `||=` operator here? This is the first time we're using `@review` so it should not be initialized previously.

If you want to retrieve a user's review of the restaurant or create blank one if they haven't left one yet, use the `find_or_initialize_by` method:

```ruby
@restaurant = Restaurant.find(params[:id])
@review = Review.find_or_initialize_by(restaurant: @restaurant, user: current_user)
```